tasks/preflight.yml

```diff
@@ -77,3 +79,39 @@
   register: prometheus_systemd_version
   tags:
     - skip_ansible_lint
+
+- block:
+    - name: Fail if docker labels are set but docker isn't
+      debug:
+        msg: "Labels are set, but the role isn't configured to install prometheus with docker"
+      when:
+        - not prometheus_docker_install
+        - prometheus_docker_labels != []
+
+    - name: Get Docker daemon executable path
+      shell: whereis dockerd | awk '{print $2}'
```

Comment:
Also you don't need a daemon, you need a client.

Reply:
my guess is that at some point I would probably face a dead end because of all engine that would be able to run a container so.. the thing that should be check is the capacity to build a container rather than checking if a daemon or a client is there